Modules/mmapmodule.c

```diff
@@ -708,11 +708,54 @@ mmap__sizeof__method(mmap_object *self, void *unused)
 }
 #endif
 
+#ifdef HAVE_MADVISE
+static PyObject *
+mmap_madvise_method(mmap_object *self, PyObject *args)
+{
+    int option;
+    Py_ssize_t start = 0, length;
+
+    CHECK_VALID(NULL);
+    length = self->size;
+
+    if (!PyArg_ParseTuple(args, "i|nn:madvise", &option, &start, &length)) {
+        return NULL;
+    }
+
+    if (start < 0 || start >= self->size) {
+        PyErr_SetString(PyExc_ValueError, "madvise start invalid");
+        return NULL;
+    }
+    if (length <= 0) {
+        PyErr_SetString(PyExc_ValueError, "madvise length invalid");
```

Comment:
Here as well: perhaps "madvise length must be > 0"?

Reply:
By the way, Linux madvise() allows length to be 0 (according to its manpage).

Reply:
Well, I'm not sure why one would pass a length of zero. I think it's okay to change the check to `length < 0`, though.